### ISSUE TITLE
[Adaptive Cards - Rate your experience]: User is unable to activate “Your Privacy is important to us” link control while navigating using keyboard TAB key.

### ISSUE THREAD

**Issue by Shireen2000** [microsoft/AdaptiveCards#7722]
### Target Platforms

Android

### SDK Version

105.0.1300.0 

### Application Name

Adaptive Cards

### Problem Description

**Test Environment:**
Android Version: 12
Application: Adaptive Cards Visualizer
App Version: Version 2021.10.15.1 (40331737)
Tool: Logitech Blutooth Keyboard

**Pre-requisites:**
Import Card > Choose File > Select File > Visual

**Repro Steps:**
- Navigate to the “Your Privacy is important to us” link present in the card.
- Observe the issue.

**Actual Result:**
User is unable to activate “Your Privacy is important to us” link control by Enter or Space Key while navigating to link using keyboard TAB key.

**Expected Result:**
User should able to activate “Your Privacy is important to us” link control by Enter or Space Key 

**User Experience:**
Keyboard users will face difficulty if User is unable to activate “Your Privacy is important to us” link control while navigating using keyboard TAB key.

**MAS Reference:**
[Keyboard - Liquid (microsoft.com)](https://liquid.microsoft.com/Web/Object/Read/MS.Accessibility/Requirements/02.01.01)

Have feedback to share on Bugs ? Please help fill [Trusted Tester Bug Feedback (office.com)](https://nam06.safelinks.protection.outlook.com/?url=https://forms.office.com/pages/responsepage.aspx?id%3Dv4j5cvGGr0GRqy180BHbR5dUqGQgJ9lItv-jqY8F09lURVdYTENIRkY2UzlCR0NKV1E2VUE5OVpDVy4u&data=04%7c01%7cv-apoorva%40microsoft.com%7c551850210d444be9b28308d9691e3dde%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c637656400998658942%7cUnknown%7cTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7c1000&sdata=l6w0SvVcMSvyGnJ6%2B6uSSgd3braF2dZ7/AdBrqqzIR8%3D&reserved=0) “

### Screenshots

https://user-images.githubusercontent.com/96283152/181270252-a9f6b32f-ae54-47c6-b6c9-b3ee97f041a5.mp4

### Card JSON

```JSON
{
  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
  "type": "AdaptiveCard",
  "version": "1.2",
  "body": [
    {
      "type": "ColumnSet",
      "columns": [
        {
          "type": "Column",
          "width": 2,
          "items": [
            {
              "type": "TextBlock",
              "text": "Rate your experience!",
              "weight": "bolder",
              "size": "medium"
            },
            {
              "type": "TextBlock",
              "text": "Please rate your experience! Your feedback is really appreciated and will help improve your experience in the future",
              "wrap": true
            },
            {
              "type": "ColumnSet",
              "columns": [
                {
                  "type": "Column",
                  "width": "30px",
                  "items": [
                    {
                      "type": "Image",
                      "url": "https://irisstorageaccdev.blob.core.windows.net/iris-ux/assets/images/star.png",
                      "id": "star1",
                      "isVisible": true,
                      "selectAction": {
                        "type": "Action.ToggleVisibility",
                        "title": "One star",
                        "targetElements": [
                          {
                            "elementId": "staryellow1",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow2",
                            "isVisible": false
                          },
                          {
                            "elementId": "staryellow3",
                            "isVisible": false
                          },
                          {
                            "elementId": "staryellow4",
                            "isVisible": false
                          },
                          {
                            "elementId": "staryellow5",
                            "isVisible": false
                          },
                          {
                            "elementId": "star1",
                            "isVisible": false
                          },
                          {
                            "elementId": "star2",
                            "isVisible": true
                          },
                          {
                            "elementId": "star3",
                            "isVisible": true
                          },
                          {
                            "elementId": "star4",
                            "isVisible": true
                          },
                          {
                            "elementId": "star5",
                            "isVisible": true
                          },
                          {
                            "elementId": "submit0",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit1",
                            "isVisible": true
                          },
                          {
                            "elementId": "submit2",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit3",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit4",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit5",
                            "isVisible": false
                          }
                        ]
                      }
                    },
                    {
                      "type": "Image",
                      "url": "https://irisstorageaccdev.blob.core.windows.net/iris-ux/assets/images/star-yellow.png",
                      "id": "staryellow1",
                      "isVisible": false,
                      "selectAction": {
                        "type": "Action.ToggleVisibility",
                        "title": "One star",
                        "targetElements": [
                          {
                            "elementId": "staryellow1",
                            "isVisible": false
                          },
                          {
                            "elementId": "staryellow2",
                            "isVisible": false
                          },
                          {
                            "elementId": "staryellow3",
                            "isVisible": false
                          },
                          {
                            "elementId": "staryellow4",
                            "isVisible": false
                          },
                          {
                            "elementId": "staryellow5",
                            "isVisible": false
                          },
                          {
                            "elementId": "star1",
                            "isVisible": true
                          },
                          {
                            "elementId": "star2",
                            "isVisible": true
                          },
                          {
                            "elementId": "star3",
                            "isVisible": true
                          },
                          {
                            "elementId": "star4",
                            "isVisible": true
                          },
                          {
                            "elementId": "star5",
                            "isVisible": true
                          },
                          {
                            "elementId": "submit0",
                            "isVisible": true
                          },
                          {
                            "elementId": "submit1",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit2",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit3",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit4",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit5",
                            "isVisible": false
                          }
                        ]
                      }
                    }
                  ]
                },
                {
                  "type": "Column",
                  "width": "30px",
                  "items": [
                    {
                      "type": "Image",
                      "url": "https://irisstorageaccdev.blob.core.windows.net/iris-ux/assets/images/star.png",
                      "id": "star2",
                      "isVisible": true,
                      "selectAction": {
                        "type": "Action.ToggleVisibility",
                        "title": "Two star",
                        "targetElements": [
                          {
                            "elementId": "staryellow1",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow2",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow3",
                            "isVisible": false
                          },
                          {
                            "elementId": "staryellow4",
                            "isVisible": false
                          },
                          {
                            "elementId": "staryellow5",
                            "isVisible": false
                          },
                          {
                            "elementId": "star1",
                            "isVisible": false
                          },
                          {
                            "elementId": "star2",
                            "isVisible": false
                          },
                          {
                            "elementId": "star3",
                            "isVisible": true
                          },
                          {
                            "elementId": "star4",
                            "isVisible": true
                          },
                          {
                            "elementId": "star5",
                            "isVisible": true
                          },
                          {
                            "elementId": "submit0",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit1",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit2",
                            "isVisible": true
                          },
                          {
                            "elementId": "submit3",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit4",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit5",
                            "isVisible": false
                          }
                        ]
                      }
                    },
                    {
                      "type": "Image",
                      "url": "https://irisstorageaccdev.blob.core.windows.net/iris-ux/assets/images/star-yellow.png",
                      "id": "staryellow2",
                      "isVisible": false,
                      "selectAction": {
                        "type": "Action.ToggleVisibility",
                        "title": "Two star",
                        "targetElements": [
                          {
                            "elementId": "staryellow1",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow2",
                            "isVisible": false
                          },
                          {
                            "elementId": "staryellow3",
                            "isVisible": false
                          },
                          {
                            "elementId": "staryellow4",
                            "isVisible": false
                          },
                          {
                            "elementId": "staryellow5",
                            "isVisible": false
                          },
                          {
                            "elementId": "star1",
                            "isVisible": false
                          },
                          {
                            "elementId": "star2",
                            "isVisible": true
                          },
                          {
                            "elementId": "star3",
                            "isVisible": true
                          },
                          {
                            "elementId": "star4",
                            "isVisible": true
                          },
                          {
                            "elementId": "star5",
                            "isVisible": true
                          },
                          {
                            "elementId": "submit0",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit1",
                            "isVisible": true
                          },
                          {
                            "elementId": "submit2",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit3",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit4",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit5",
                            "isVisible": false
                          }
                        ]
                      }
                    }
                  ]
                },
                {
                  "type": "Column",
                  "width": "30px",
                  "items": [
                    {
                      "type": "Image",
                      "url": "https://irisstorageaccdev.blob.core.windows.net/iris-ux/assets/images/star.png",
                      "id": "star3",
                      "isVisible": true,
                      "selectAction": {
                        "type": "Action.ToggleVisibility",
                        "title": "Three star",
                        "targetElements": [
                          {
                            "elementId": "staryellow1",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow2",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow3",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow4",
                            "isVisible": false
                          },
                          {
                            "elementId": "staryellow5",
                            "isVisible": false
                          },
                          {
                            "elementId": "star1",
                            "isVisible": false
                          },
                          {
                            "elementId": "star2",
                            "isVisible": false
                          },
                          {
                            "elementId": "star3",
                            "isVisible": false
                          },
                          {
                            "elementId": "star4",
                            "isVisible": true
                          },
                          {
                            "elementId": "star5",
                            "isVisible": true
                          },
                          {
                            "elementId": "submit0",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit1",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit2",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit3",
                            "isVisible": true
                          },
                          {
                            "elementId": "submit4",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit5",
                            "isVisible": false
                          }
                        ]
                      }
                    },
                    {
                      "type": "Image",
                      "url": "https://irisstorageaccdev.blob.core.windows.net/iris-ux/assets/images/star-yellow.png",
                      "id": "staryellow3",
                      "isVisible": false,
                      "selectAction": {
                        "type": "Action.ToggleVisibility",
                        "title": "Three star",
                        "targetElements": [
                          {
                            "elementId": "staryellow1",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow2",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow3",
                            "isVisible": false
                          },
                          {
                            "elementId": "staryellow4",
                            "isVisible": false
                          },
                          {
                            "elementId": "staryellow5",
                            "isVisible": false
                          },
                          {
                            "elementId": "star1",
                            "isVisible": false
                          },
                          {
                            "elementId": "star2",
                            "isVisible": false
                          },
                          {
                            "elementId": "star3",
                            "isVisible": true
                          },
                          {
                            "elementId": "star4",
                            "isVisible": true
                          },
                          {
                            "elementId": "star5",
                            "isVisible": true
                          },
                          {
                            "elementId": "submit0",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit1",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit2",
                            "isVisible": true
                          },
                          {
                            "elementId": "submit3",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit4",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit5",
                            "isVisible": false
                          }
                        ]
                      }
                    }
                  ]
                },
                {
                  "type": "Column",
                  "width": "30px",
                  "items": [
                    {
                      "type": "Image",
                      "url": "https://irisstorageaccdev.blob.core.windows.net/iris-ux/assets/images/star.png",
                      "id": "star4",
                      "isVisible": true,
                      "selectAction": {
                        "type": "Action.ToggleVisibility",
                        "title": "Four star",
                        "targetElements": [
                          {
                            "elementId": "staryellow1",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow2",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow3",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow4",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow5",
                            "isVisible": false
                          },
                          {
                            "elementId": "star1",
                            "isVisible": false
                          },
                          {
                            "elementId": "star2",
                            "isVisible": false
                          },
                          {
                            "elementId": "star3",
                            "isVisible": false
                          },
                          {
                            "elementId": "star4",
                            "isVisible": false
                          },
                          {
                            "elementId": "star5",
                            "isVisible": true
                          },
                          {
                            "elementId": "submit0",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit1",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit2",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit3",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit4",
                            "isVisible": true
                          },
                          {
                            "elementId": "submit5",
                            "isVisible": false
                          }
                        ]
                      }
                    },
                    {
                      "type": "Image",
                      "url": "https://irisstorageaccdev.blob.core.windows.net/iris-ux/assets/images/star-yellow.png",
                      "id": "staryellow4",
                      "isVisible": false,
                      "selectAction": {
                        "type": "Action.ToggleVisibility",
                        "title": "Four star",
                        "targetElements": [
                          {
                            "elementId": "staryellow1",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow2",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow3",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow4",
                            "isVisible": false
                          },
                          {
                            "elementId": "staryellow5",
                            "isVisible": false
                          },
                          {
                            "elementId": "star1",
                            "isVisible": false
                          },
                          {
                            "elementId": "star2",
                            "isVisible": false
                          },
                          {
                            "elementId": "star3",
                            "isVisible": false
                          },
                          {
                            "elementId": "star4",
                            "isVisible": true
                          },
                          {
                            "elementId": "star5",
                            "isVisible": true
                          },
                          {
                            "elementId": "submit0",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit1",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit2",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit3",
                            "isVisible": true
                          },
                          {
                            "elementId": "submit4",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit5",
                            "isVisible": false
                          }
                        ]
                      }
                    }
                  ]
                },
                {
                  "type": "Column",
                  "width": "30px",
                  "items": [
                    {
                      "type": "Image",
                      "url": "https://irisstorageaccdev.blob.core.windows.net/iris-ux/assets/images/star.png",
                      "id": "star5",
                      "isVisible": true,
                      "selectAction": {
                        "type": "Action.ToggleVisibility",
                        "title": "Five star",
                        "targetElements": [
                          {
                            "elementId": "staryellow1",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow2",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow3",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow4",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow5",
                            "isVisible": true
                          },
                          {
                            "elementId": "star1",
                            "isVisible": false
                          },
                          {
                            "elementId": "star2",
                            "isVisible": false
                          },
                          {
                            "elementId": "star3",
                            "isVisible": false
                          },
                          {
                            "elementId": "star4",
                            "isVisible": false
                          },
                          {
                            "elementId": "star5",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit0",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit1",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit2",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit3",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit4",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit5",
                            "isVisible": true
                          }
                        ]
                      }
                    },
                    {
                      "type": "Image",
                      "url": "https://irisstorageaccdev.blob.core.windows.net/iris-ux/assets/images/star-yellow.png",
                      "id": "staryellow5",
                      "isVisible": false,
                      "selectAction": {
                        "type": "Action.ToggleVisibility",
                        "title": "Five star",
                        "targetElements": [
                          {
                            "elementId": "staryellow1",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow2",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow3",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow4",
                            "isVisible": true
                          },
                          {
                            "elementId": "staryellow5",
                            "isVisible": false
                          },
                          {
                            "elementId": "star1",
                            "isVisible": false
                          },
                          {
                            "elementId": "star2",
                            "isVisible": false
                          },
                          {
                            "elementId": "star3",
                            "isVisible": false
                          },
                          {
                            "elementId": "star4",
                            "isVisible": false
                          },
                          {
                            "elementId": "star5",
                            "isVisible": true
                          },
                          {
                            "elementId": "submit0",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit1",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit2",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit3",
                            "isVisible": false
                          },
                          {
                            "elementId": "submit4",
                            "isVisible": true
                          },
                          {
                            "elementId": "submit5",
                            "isVisible": false
                          }
                        ]
                      }
                    }
                  ]
                }
              ]
            },
            {
              "type": "TextBlock",
              "text": "Feedback Category",
              "weight": "bolder",
              "wrap": true
            },
            {
              "type": "Input.ChoiceSet",
              "id": "category",
              "style": "compact",
              "isMultiSelect": false,
              "choices": []
            },
            {
              "type": "TextBlock",
              "text": "Additional Comments / Suggestions",
              "weight": "bolder",
              "wrap": true
            },
            {
              "type": "TextBlock",
              "text": "Please do not include any personal or confidential information in your comment",
              "wrap": true
            },
            {
              "type": "Input.Text",
              "placeholder": "Leave a comment or suggestion",
              "isMultiline": true,
              "id": "comment"
            },
            {
              "type": "Input.Toggle",
              "id": "contactMe",
              "title": "You can contact me about this feedback",
              "value": "false",
              "valueOn": "true",
              "valueOff": "false"
            },
            {
              "type": "Input.Text",
              "placeholder": "Email (Optional)",
              "id": "email"
            },
            {
              "type": "TextBlock",
              "text": "[Your privacy is important to us](https://go.microsoft.com/fwlink/?LinkId=518021)",
              "size": "small"
            },
            {
              "type": "ActionSet",
              "id": "submit0",
              "isVisible": true,
              "actions": [
                {
                  "type": "Action.Submit",
                  "title": "Submit",
                  "data": { "rating": 0 }
                }
              ]
            },
            {
              "type": "ActionSet",
              "id": "submit1",
              "isVisible": false,
              "actions": [
                {
                  "type": "Action.Submit",
                  "title": "Submit",
                  "data": { "rating": 1 }
                }
              ]
            },
            {
              "type": "ActionSet",
              "id": "submit2",
              "isVisible": false,
              "actions": [
                {
                  "type": "Action.Submit",
                  "title": "Submit",
                  "data": { "rating": 2 }
                }
              ]
            },
            {
              "type": "ActionSet",
              "id": "submit3",
              "isVisible": false,
              "actions": [
                {
                  "type": "Action.Submit",
                  "title": "Submit",
                  "data": { "rating": 3 }
                }
              ]
            },
            {
              "type": "ActionSet",
              "id": "submit4",
              "isVisible": false,
              "actions": [
                {
                  "type": "Action.Submit",
                  "title": "Submit",
                  "data": { "rating": 4 }
                }
              ]
            },
            {
              "type": "ActionSet",
              "id": "submit5",
              "isVisible": false,
              "actions": [
                {
                  "type": "Action.Submit",
                  "title": "Submit",
                  "data": { "rating": 5 }
                }
              ]
            }
          ]
        }
      ]
    }
  ]
}
```

**Comment by anna-dingler** [comment]
Hi @Shireen2000, 

Is this a duplicate of https://github.com/microsoft/AdaptiveCards/issues/7194, or is there a different issue here?

The issue linked above was resolved on PR #7554.

Thanks!

**Comment by anna-dingler** [comment]
Hi @Shireen2000,

I am going to close this issue per the comment above. I also tested locally and was unable to reproduce the error.

Let me know if you have any questions. Thanks!
